Packages/Tracking/Interaction Engine/Runtime/Scripts/Internal/ContactBone.cs

```diff
@@ -126,13 +129,37 @@ void OnDestroy()
             interactionController.manager.contactBoneBodies.Remove(rigidbody);
         }
 
+        private void FixedUpdate()
+        {
+            foreach (var contacting in contactingInteractionBehaviours)
+            {
+                if (!contactingStayInteractionBehaviours.Contains(contacting.Key))
+                {
+                    interactionController.NotifyContactBoneCollisionExit(this, contacting.Key);
+
+                    ContactingObjectsToRemove.Add(contacting.Key);
+                }
+            }
+
+            foreach(var removed in ContactingObjectsToRemove)
+            {
+                contactingInteractionBehaviours.Remove(removed);
+            }
```

Comment:
Is populating a 'to remove' list necessary here? can we remove them in the first loop if it's a `for` loop that iterates backwards through the `contactingStayInteractionBehaviours` list instead?